app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivityPresenter.kt

```diff
@@ -86,10 +91,47 @@ class AdministratorControlsActivityPresenter @Inject constructor(
     getAdministratorControlsFragment()!!.setSelectedFragment(lastLoadedFragment)
     activity.supportFragmentManager.beginTransaction().add(
       R.id.administrator_controls_fragment_multipane_placeholder,
-      AppVersionFragment()
+      AppVersionFragment(),
+      ADMINISTRATOR_CONTROLS_BACKSTACK
     ).commitNow()
   }
 
+  fun loadProfileEdit(profileId: Int) {
+    lastLoadedFragment = PROFILE_EDIT_FRAGMENT
+    binding.administratorControlsMultipaneOptionsBackButton!!.visibility = View.VISIBLE
+    val fragment = ProfileEditFragment.newInstance(profileId, isMultipane)
+    activity.supportFragmentManager.beginTransaction().add(
+      R.id.administrator_controls_fragment_multipane_placeholder,
+      fragment,
+      ADMINISTRATOR_CONTROLS_BACKSTACK
+    )
+      .addToBackStack(ADMINISTRATOR_CONTROLS_BACKSTACK)
+      .commit()
+  }
+
+  fun handleOnBackPressed() {
+    if (activity.supportFragmentManager.backStackEntryCount > 0)
+      activity.supportFragmentManager.popBackStackImmediate()
```

Comment:
Per my comment from earlier, what alternative approaches have you considered vs. using the backstack?

@rt4914 do you have any thoughts here?

Reply:
Other approaches are manually managing the back stack by storing which fragment is on Top which we usally do in oppia we have to store all the tags of the fragments.
But by using back stack we just have to check on the count of fragments which is better and we can add or pop the fragments from the backstack at ease
we can even store the state of the back stack like we do in instagram and with backstack we can also implement the back button much easier and a nicer way.

Reply:
Per the other discussion, the implementation was updated to no longer commit to the backstack manually.